### PR TITLE
fix: remove ReactElement casting from render

### LINKED ISSUE #145: 'ErrorBoundary' cannot be used as a JSX component.
- `react-error-boundary` version: 4.0.8
- `node` version: latest
- `npm` version: latest

Relevant code or config

```js
import { ErrorBoundary } from 'react-error-boundary';

function App() {
  return (
    <ErrorBoundary fallback={<div>Something went wrong</div>}>
      <h1>Hello</h1>
    </ErrorBoundary>
  );
}
```

What you did: Installed this package and started using it.

What happened: The app stopped compiling.

Reproduction repository: https://stackblitz.com/edit/stackblitz-starters-d6a6ck?file=src%2FApp.tsx

Problem description: Wrong types?

Suggested solution: IDK, but I've seen this is a recurring issue here.

### PR DESCRIPTION
**Pull request by KubaJastrz**
closes #145

**What**:

- Remove type casting in ErrorBoundary.prototype.render
- ~~Remove outdated warning in README~~

**Why**:

For whatever reason (I didn't want to invest much time into debugging React types), `ReactElement` returned from class component render function is not compatible with JSX declarations.

My change is removing the explicit type, which makes it fall back to the inferred type
```ts
ErrorBoundary.render(): React.FunctionComponentElement<React.ProviderProps<ErrorBoundaryContextType | null>>
```

I've seen this commit 63339c36f7ff6e16a9648f4d0afea085cb69bb2a, which added `as ReactElement` in the first place. However, after short investigation, I noticed that the distributed files don't suffer from the original problem of broken import paths. So I assumed it is safe to roll it back to previous state.

**How**:

?

**Checklist**:

- [x] Documentation
- [ ] Tests
- [x] Ready to be merged

### CODE REVIEW

File: src/ErrorBoundary.ts [hunk 1]
```diff
@@ -112,7 +112,7 @@ export class ErrorBoundary extends Component<
         },
       },
       childToRender
-    ) as ReactElement;
+    );
```
Comment (bvaughn) [inline src/ErrorBoundary.ts:115]: Note that the change you're removing here was added to address issue #133 so I don't know of landing this change would regress that one.

File: README.md [hunk 2]
```diff
@@ -180,32 +180,4 @@ const ComponentWithErrorBoundary = withErrorBoundary(ExampleComponent, {
...
-  ...
-  "resolutions": {
-    "@types/react": "17.0.60"
-  },
-  ...
-}
-```
-
----
-
```
Comment (bvaughn) [inline README.md:210]: I think this is still worth keeping for now, as this _does_ seem to be another potential issue based on @fh-igor-serko's comment.

File: src/ErrorBoundary.ts (hunk 1 above)
Comment (KubaJastrz) [reply src/ErrorBoundary.ts:115]: _"it worked on my machine"_

Comment (bvaughn) [reply src/ErrorBoundary.ts:115]: You should be able to run `pnpm prerelease` locally and verify the output fwiw.

Comment (KubaJastrz) [reply src/ErrorBoundary.ts:115]: I did that. Posted the output above.

I wanted to make sure it's the same for you too.

Comment (bvaughn) [reply src/ErrorBoundary.ts:115]: Oh okay gotcha. I'll try to check this out sometime this afternoon. 👍🏼

**Comment by bvaughn** [comment]
Okay. Now that the work day has ended and I'm able to focus on this for more than 30 seconds, I reminded myself of what the root cause of #133 was– a Parcel bug that messed up imports for some reason. Not one of the subtle TypeScript+React bugs that have been cropping up for me lately. (I guess I just assumed it would be another of those.)

It looks like Preconstruct handles the return type better, though still a bit verbose:
```ts
render(): import("react").FunctionComponentElement<
  import("react").ProviderProps<
    import("./ErrorBoundaryContext.js").ErrorBoundaryContextType | null
  >
>;
```

I guess this change seems like a good one to land then 👍🏼 Thanks

**Comment by bvaughn** [comment]
Released as 4.0.9

**Pull request merged**